cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go

```diff
@@ -695,20 +696,33 @@ func resourceRefsFromManifest(m string) ([]*corev1.ResourceRef, error) {
 		}
 		if doc.Kind == "List" || doc.Kind == "RoleList" || doc.Kind == "ClusterRoleList" {
 			for _, i := range doc.Items {
+				namespace := i.Metadata.Namespace
+				if namespace == "" {
+					namespace = pkgNamespace
+				}
 				refs = append(refs, &corev1.ResourceRef{
 					ApiVersion: i.APIVersion,
 					Kind:       i.Kind,
 					Name:       i.Metadata.Name,
-					Namespace:  i.Metadata.Namespace,
+					Namespace:  namespace,
 				})
 			}
 			continue
 		}
+		// Helm does not require that the rendered manifest specifies the
+		// resource namespace so some charts do not do so (ldap).  We explicitly
+		// set the namespace for the resource ref so that it can be used as part
+		// of the key for the resource ref.  At the moment we do not distinguish
+		// between cluster-scoped and namespace-scoped resources for the refs.
```

Comment:
This came up while using the new plugin from the dashboard: the LDAP chart just happens to not specify the namespace metadata in the rendered templates of the helm manifest, which meant that all the resource refs were missing this info, which caused problems on two levels:
* The resource refs are meant to additionally check that a specific resource belongs to a package... if it does not include the namespace, it could be used to read a similarly named resource (even a secret) in a different namespace.
* In the dashboard code, we need to store resources with a unique key in the redux state, which of course needs to include the resource. This was what found the issue for me since the corresponding code which was pulling data out of the redux state wouldn't find it (since the original key did not have the namespace).

Reply:
So, just to confirm I'm getting it right: even if packages can generate cluster-scoped resources, we are not supporting it?  That is, are we assuming a "no namespace" = "namespace where the pkg in installed" ? 
If so, how are we retrieving those cluster-scoped resources? Perhaps it's just me missing something

Reply:
Cluster-scoped resources are fine, in that when fetching the referenced resource in the resources plugin, it checks whether the resource is namespaced via the k8s rest mapper and does the right thing (ie. it does not rely on the namespace to determine the scope)... rather than just relying on `ref.Namespace` (this is a change below).

The issue is that, as we're using the helm manifest to get the references, it is decided by the chart author, so you get (for example) both `ClusterRole`s with a `metadata.namespace` set (which, when applied, just results in the cluster-scoped resource) and `Role`s without a namespace (as Helm automatically applies the resources in the context of the release namespace). I'm just standardising here so the generated ResourceRef(erence) will always have the ResourceRef.Namespace set, regardless of the scope of the resource so that (a) we can define a unique key/identifier for a resource ref, and (b) the reference cannot be used to get/watch resources in other namespaces. 

Ideally here we'd only be setting the namespace only when it is not set *and* the resource is a namespace-scoped resource, which is what I'm referring to in the last sentence of the above comment, but that means providing both plugins with a RESTMapper, which I didn't do here. I'll update that to an explicit TODO.